docs/user-guide/horizontal-pod-autoscaler.md

```diff
@@ -0,0 +1,126 @@
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/user-guide/horizontal-pod-autoscaler.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Horizontal Pod Autoscaler
+
+This document describes the current state of Horizontal Pod Autoscaler in Kubernetes.
+
+<!-- BEGIN MUNGE: GENERATED_TOC -->
+
+- [Horizontal Pod Autoscaler](#horizontal-pod-autoscaler)
+  - [What is Horizontal Pod Autoscaler?](#what-is-horizontal-pod-autoscaler)
+  - [How does Horizontal Pod Autoscaler work?](#how-does-horizontal-pod-autoscaler-work)
+  - [API Object](#api-object)
+  - [Support for horizontal pod autoscaler in kubectl](#support-for-horizontal-pod-autoscaler-in-kubectl)
+  - [Autoscaling during rolling update](#autoscaling-during-rolling-update)
+  - [Further reading](#further-reading)
+
+<!-- END MUNGE: GENERATED_TOC -->
+
+
+## What is Horizontal Pod Autoscaler?
+
+Horizontal pod autoscaling allows the number of pods in a replication controller or deployment
+to scale automatically based on observed CPU utilization.
+It is a [beta](../api.md#api-versioning) feature in Kubernetes 1.1.
+
+The autoscaler is implemented as a Kubernetes API resource and a controller.
+The resource describes behavior of the controller.
+The controller periodically adjusts the number of replicas in replication controller or deployment
+to match the observed average CPU utilization to the target specified by user.
+
+
+## How does Horizontal Pod Autoscaler work?
+
+![Horizontal Pod Autoscaler diagram](horizontal-pod-autoscaler.png)
+
+The autoscaler is implemented as a control loop.
+It periodically queries CPU utilization of pods it is responsible for.
+(The period of the autoscaler is controlled by `--horizontal-pod-autoscaler-sync-period` flag of controller manager.
+The default value is 30 seconds).
+Then, it compares the arithmetic mean of the pods' CPU utilization with the target and adjust the number of replicas if needed.
+
+CPU utilization is the recent CPU usage of a pod divided by the sum of CPU requested by the pod's containers.
+Please note that if some of the pod's containers do not have CPU request set,
+CPU utilization for the pod will not be defined and the autoscaler will not take any action.
```

Comment:
Do we throw an event when this happens or anything that the user can see when they get in this state?  I think it will be a common error.